src/cartridge/mod.rs

```diff
@@ -80,11 +95,17 @@ impl Cartridge {
         }
     }
 
-    pub fn load_rom(&mut self, path: String) -> Result<(),io::Error> {
+    pub fn load_rom(&mut self, path: String) -> Result<(), io::Error> {
         self.rom = fs::read(path)?;
-        self.loaded = true;
+        if self.rom.len() < (HEADER_START + HEADER_SIZE) {
+            panic!("missing header");
```

Comment:
Ya I agree. Just didn't do this yet because I'm not really sure what the best way to handle program exiting is in rust. The official ways described by the maintainers is pretty dookie (but in a bad way, not a good way like dookieboy)